openmdao/recorders/tests/test_sqlite_reader.py

```diff
@@ -2342,6 +2342,52 @@ def test_feature_driver_options_with_values(self):
         self.assertEqual((objs['obj'], objs['obj_cmp.obj']), (objs['obj_cmp.obj'], objs['obj']))
         self.assertEqual((dvs['x'], dvs['px.x']), (dvs['px.x'], dvs['x']))
 
+    def test_feature_list_inputs_and_outputs(self):
+        import openmdao.api as om
+        from openmdao.test_suite.components.sellar import SellarProblem
+
+        import numpy as np
+        prob = SellarProblem()
+
+        recorder = om.SqliteRecorder("cases.sql")
+        prob.model.add_recorder(recorder)
+        prob.model.recording_options['record_residuals'] = True
+
+        prob.setup()
+
+        d1 = prob.model.d1  # SellarDis1withDerivatives (an ExplicitComp)
+        d1.nonlinear_solver = om.NonlinearBlockGS(maxiter=5)
+        d1.add_recorder(recorder)
+
+        prob.run_driver()
+        prob.cleanup()
+
+        cr = om.CaseReader("cases.sql")
+
+        expected_inputs_case = {
+            'd1.z': {'value': [5., 2.]},
+            'd1.x': {'value': [1.]},
+            'd1.y2': {'value': [12.27257053]}
+        }
+        expected_outputs_case = {
+            'd1.y1': {'value': [25.545485893882876]},
+        }
+
+        system_cases = cr.list_cases('root.d1')
+
+        case = cr.get_case(system_cases[1])
+
+        inputs_case = sorted( case.list_inputs(values=True, units=False, hierarchical=True,
+                                       out_stream=None))
```

Comment:
The `units`  and `hierarchical` options here are extraneous.. none of the vars have units and nothing is being printed.

Maybe for the docs we want to let it print, so leave `hierarchical` and remove the `out_stream` arg?  Then we wouldn't need the unused `expected` vars (above) to understand what the following assert/print statements are showing.